tests/fixtures/inbox_data.py

```diff
@@ -202,7 +215,31 @@ def setup_data(desired_wells):
 
         return desired_wells
 
-    return setup_data
+    yield setup_data
+
+    print("\nTEARDOWN test_data_factory")
+
+    mlwhdb_test_session.execute(text("SET foreign_key_checks=0;"))
```

Comment:
Same as before. If it is used only here there shouldn't be problems of performance I think.

Reply:
Foreign key constraint is dropped to allow to delete datat from table in arbitrary order. In general in tests we want to have this constraint on.